crates/ruff_linter/src/settings/mod.rs

```diff
@@ -38,13 +41,34 @@ pub mod flags;
 pub mod rule_table;
 pub mod types;
 
+#[macro_export]
+macro_rules! display_settings {
+    (formatter = $fmt:ident, namespace = $namespace:literal, fields = [$($settings:ident.$field:ident $(| $modifier:tt)?),* $(,)?]) => {
```

Comment:
I might be tempted to rename `namespace` to `prefix`, since it seems to usually end with a `.`. (This is a small thing.)

Reply:
Or I wonder, could the `.` be inserted unconditionally in this macro?

I would guess maybe not, for cases where the namespace is empty?

Reply:
Alternatively, we could make the namespace/prefix an optional parameter and then append `.` if a namespace is specified.

Reply:
I defer to you on what to do here. :)